src/sentry/search/events/datasets/discover.py

```diff
@@ -2002,3 +2005,15 @@ def _error_handled_filter_converter(
 
     def _key_transaction_filter_converter(self, search_filter: SearchFilter) -> WhereType | None:
         return filter_aliases.team_key_transaction_filter(self.builder, search_filter)
+
+    def _event_type_filter_converter(self, search_filter: SearchFilter) -> WhereType | None:
+        if self.builder.dataset == Dataset.Transactions:
+            if search_filter.operator in ["=", "IN"] and search_filter.value.value in [
+                "transaction",
+                ["transaction"],
+            ]:
+                return Condition(Function("equals", [1, 1]), Op.EQ, 1)
```

Comment:
No-op on event type filter with the transactions dataset since the field doesn't exist on the transactions table. right now a condition like `event.type:transaction` get converted into a tag condition

Reply:
🤔 does returning None not work here? (legit forget, but i thought we just ignore the condition in that case)

Reply:
oh nice, that works

Reply:
hmm actually, for an OR condition (for example, `project:foo OR event.type:transaction`), we'll want to no-op instead of setting to None because setting to None will evaluate to `project:foo` which is a more restrictive condition than `project:foo OR event.type:transaction`)

Reply:
my example _is_ as really weird condition though, and would probably evaluate as an ambiguous/"discover" dataset when we try to determine the dataset, so I think I'll return None here for now